test/rules/experiments/nnbd/rules/prefer_constructors_over_static_methods.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_constructors_over_static_methods`
+
+class A {
+  static final array = <A>[];
+
+  A.internal();
+
+  static A bad1() => // LINT
+      new A.internal();
+
+  static A get newA => // LINT
+      new A.internal();
+
+  static A bad2() { // LINT
+    final a = new A.internal();
+    return a;
+  }
+
+  static A good1(int i) { // OK
+    return array[i];
+  }
+
+  factory A.good2() { // OK
```

Comment:
I would have assumed that this lint never flags constructors, so I'm not sure what value this (or the next) test adds.

Reply:
Agreed.  These tests were written loooonnnnggg ago and contributed by Juan.  Not sure what the rationale was?

Reply:
Me either. But that shouldn't stop us from cleaning it up.